rest/config.go

```diff
@@ -266,6 +267,55 @@ func GetTLSVersionFromString(stringV *string) uint16 {
 	return uint16(DefaultMinimumTLSVersionConst)
 }
 
+type invalidConfigInfo struct {
+	logged              bool
+	configBucketName    string
+	persistedBucketName string
+}
+
+type invalidDatabaseConfigs struct {
+	dbNames    map[string]*invalidConfigInfo
+	loggingCtx context.Context
+	m          sync.RWMutex
+}
+
+// addInvalidDatabase adds a db to invalid dbconfig map if it doesn't exist in there yet and will log for it at warning level
+// if the db already exists there we will calculate if we need to log again according to the config update interval
+func (d *invalidDatabaseConfigs) addInvalidDatabase(dbname string, refreshInterval *base.ConfigDuration, cnf DatabaseConfig, bucket string) {
+	configInfo := invalidConfigInfo{
+		configBucketName:    *cnf.Bucket,
+		persistedBucketName: bucket,
+	}
+	d.m.Lock()
+	defer d.m.Unlock()
+	if d.dbNames[dbname] == nil {
+		// db hasn't been tracked as invalid config yet so add it
+		d.dbNames[dbname] = &configInfo
+	}
+	// if we get here we already have the db logged as an invalid config, so now we need to work out iof we should log for it now
+	if !d.dbNames[dbname].logged {
+		// we need to log at warning if we get here
+		base.WarnfCtx(d.loggingCtx, "Mismatch in database config for database %s bucket name: %s and backend bucket: %s You must update database config immediately", dbname, d.dbNames[dbname].configBucketName, d.dbNames[dbname].persistedBucketName)
```

Comment:
I think we can make this comment more clear:

"The database config for %q in bucket %q points to %q as the bucket in the configuration will. You must update database config immediately", base.MD(dbname, d.dbNames[dbname].persistedBucketName), (d.dbNames[dbname].configBucketName)

I'd prefer to update this function to only have this string once since it would be easy to find it one place and not the other. We should add base.MD() since this is technically redactable information, though I don't believe we redact this at this time. I haven't tried this but you might be able to declare this `[]fmt.Stringer{"msg", base.MD("foo"), }

We could almost put the logging function as a function pointer to deduplicate but WarnfCtx takes fewer arguments than InfofCtx